lib/pubsub_client.rb

```diff
@@ -1,48 +1,50 @@
+# frozen_string_literal: true
+
 require 'pubsub_client/version'
 require 'pubsub_client/null_publisher_factory'
 require 'pubsub_client/null_subscriber_factory'
 require 'pubsub_client/publisher_factory'
 require 'pubsub_client/subscriber_factory'
 
-module PubsubClient
+class PubsubClient
   Error = Class.new(StandardError)
   ConfigurationError = Class.new(Error)
   CredentialsError = Class.new(Error)
   InvalidTopicError = Class.new(Error)
   InvalidSubscriptionError = Class.new(Error)
 
-  class << self
-    def stub!
-      raise ConfigurationError, 'PubsubClient is already configured' if @publisher_factory || @subscriber_factory
-      @publisher_factory = NullPublisherFactory.new
-      @subscriber_factory = NullSubscriberFactory.new
-      @stubbed = true
-    end
+  attr_reader :stubbed, :publisher_factory, :subscriber_factory
```

Comment:
These can all be private attributes as applications using this should not be modifying any of these.